bool enable
int32 meters
---
